### DISCUSSION TITLE
Is there an event I can use after composer.json is completely written to when requiring new packages?

### DISCUSSION THREAD

**Discussion by defunctl** [composer/composer#12047] [Q&A / Support] [answered]
I'm looking to automatically copy the composer.json file elsewhere after any changes happen to it.

For example, when you require certain packages they are initially written to composer.json with `"vendor/name": "*"`, and it's not until the entire process is complete that it will have the proper versioning in composer.json, e.g. `"vendor/name": "^2.0"`. 

Using `post-update-cmd`, the copied composer.json contains  `"vendor/name": "*"` and not  `"vendor/name": "^2.0"`.

Composer version: 2.7.7.

**Comment by defunctl** [comment, marked as answer]
I worked around this for now by having my static class execute all of its code inside of a callback that's passed to [register_shutdown_function](https://www.php.net/register_shutdown_function)

**Reply by Seldaek** [reply]
Yes that's a decent solution, as there is indeed no post-require-cmd or post-command-run to hook into.
